### ISSUE TITLE
ether_sky - In OCL_ZVE, we don't take into account the tokens that already exist when adding liquidity to the Uniswap V2 pool

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2024-03-zivoe-judging#388]
ether_sky

medium

# In OCL_ZVE, we don't take into account the tokens that already exist when adding liquidity to the Uniswap V2 pool

## Summary
To deposit `liquidity` into the `Uniswap V2 pool` through `OCL_ZVE`, the `DAO` sends both tokens to this contract.
If the actual deposited amount is less than the minimum allowed amounts, the deposit will be reverted.
The minimum allowed amounts are determined based on the current `token balance` of `OCL_ZVE`.
The current `token balance` includes the amounts transferred by the `DAO` for the deposit, as well as any existing amounts that can be donated by anyone or remaining from previous operations.
## Vulnerability Detail
The `pushToLockerMulti` function can only be called by the `DAO`.
```solidity
function pushToLockerMulti(
    address[] calldata assets, uint256[] calldata amounts, bytes[] calldata data
) external override onlyOwner nonReentrant {
}
```
And the `DAO` can call this function only after the `voting process`.
This operation requires time and effort, so the reversal of this operation is not good.
The `DAO` sends both tokens to the `OCL_ZVE` for the deposit, and we deposit all the current `balance` into the `Uniswap V2 pool`.
```solidity
function pushToLockerMulti(
    address[] calldata assets, uint256[] calldata amounts, bytes[] calldata data
) external override onlyOwner nonReentrant {
    for (uint256 i = 0; i < 2; i++) {
        require(amounts[i] >= 10 * 10**6, "OCL_ZVE::pushToLockerMulti() amounts[i] < 10 * 10**6");
        IERC20(assets[i]).safeTransferFrom(owner(), address(this), amounts[i]);  // @audit, here
    }
    uint balPairAsset = IERC20(pairAsset).balanceOf(address(this));
    uint balZVE = IERC20(ZVE).balanceOf(address(this));
    IERC20(pairAsset).safeIncreaseAllowance(router, balPairAsset);
    IERC20(ZVE).safeIncreaseAllowance(router, balZVE);

    (uint256 depositedPairAsset, uint256 depositedZVE, uint256 minted) = IRouter_OCL_ZVE(router).addLiquidity(
        pairAsset, 
        ZVE, 
        balPairAsset,  // @audit, here
        balZVE,  // @audit, here
        (balPairAsset * 9) / 10,
        (balZVE * 9) / 10, 
        address(this), block.timestamp + 14 days
    );
```
The actual deposited amount is based on the current `reserves` of the `pool`, so we specify the `minimum allowed amounts`.
If the actual deposited amounts are less than these values, the transaction will be reverted.

Here, the `minimum allowed amounts` include the already existing amounts in the `OCL_ZVE`.
These values can be easily manipulated by anyone or accidently by previous remaining amounts.
For example, if the current `reserves` are `10,000` for both tokens, and the `DAO` wants to deposit `100` for both tokens.
If there are `20 ZVE` tokens in the `OCL_ZVE`(or donated by a malicious user), the actual deposited amount will be less than the minimum allowed amounts, resulting in the deposit being reverted.
```solidity
actual deposited ZVE amount = 100
minimum allowed ZVE amounts = (100 + 20) * 9 / 10 = 108
```

When a `proposal` is made, nobody can anticipate the actual `balance` at the time of the future deposit.
Even if we ignore the donated amounts by malicious users, can be be certain that the `proposers` always consider the already existing amounts in the `OCL_ZVE`?
## Code Snippet
https://github.com/sherlock-audit/2024-03-zivoe/blob/d4111645b19a1ad3ccc899bea073b6f19be04ccd/zivoe-core-foundry/src/lockers/OCL/OCL_ZVE.sol#L172-L174
https://github.com/sherlock-audit/2024-03-zivoe/blob/d4111645b19a1ad3ccc899bea073b6f19be04ccd/zivoe-core-foundry/src/lockers/OCL/OCL_ZVE.sol#L192-L195
## Tool used

Manual Review

## Recommendation
We can deposit the amounts that the `DAO` sends.

Duplicate of #64
